packages/daemon/src/db/index.ts

```diff
@@ -1609,3 +1561,45 @@ export const getTokenSymbols = async (
     return prev;
   }, {}) as unknown as StringMap<string>;
 };
+
+/**
+ * Get maximum indices for multiple wallets in a single query.
+ * 
+ * @remarks
+ * This is an optimized version that combines both getMaxIndexAmongAddresses and getMaxWalletAddressIndex
+ * into a single query for multiple wallets. This reduces the number of database round trips significantly.
+ * 
+ * @param mysql - Database connection
+ * @param walletData - Array of objects containing wallet IDs and their associated addresses
+ * @returns Map of wallet IDs to their maximum indices (both among specific addresses and overall)
+ */
+export const getMaxIndicesForWallets = async (
+  mysql: MysqlConnection,
+  walletData: Array<{walletId: string, addresses: string[]}>
+): Promise<Map<string, {maxAmongAddresses: number | null, maxWalletIndex: number | null}>> => {
+  if (walletData.length === 0) {
+    return new Map();
+  }
+
+  const allAddresses = walletData.flatMap(d => d.addresses);
+  const walletIds = walletData.map(d => d.walletId);
+
+  const [results] = await mysql.query<MaxAddressIndexRow[]>(
```

Comment:
I think this query deserves better comment description. It's not so easy to understand with the group by wallets to get the max of each wallet.

Reply:
Done, can you please check again?